### PR TITLE
lib: getopt: Fix compiler warning.

### LINKED ISSUE #57520: posix: getopt: Compile warnings with getopt_long
**Describe the bug**
After the changes by https://github.com/zephyrproject-rtos/zephyr/pull/48130, lib/posix/getopt/getopt_long.c will compile with a warning that "char* dash_prefix" may be uninitialized before use in function "parse_long_options".

While the root concern of this warning will never happen on runtime due to conditional checks, the compiler is conservative and throws the warning never the less, causing problems if the use of this file is to be combined with the -Werror compile flag.

**To Reproduce**
- Set up a blank project for nRF5340
- Include getopt and call the parse_long_options function at least once in the source code
- Build using west build, and make sure to include -Werror as a compile flag:
- west build -p -b nrf5340_audio_dk_nrf5340_cpuapp -d build_output_folder -- -DCMAKE_C_FLAGS="-Werror"

**Expected behavior**
Build should be successful

**Impact**
Any project using a strict no-build-warning policy is unable to use getopt_long.c.

**Environment:**
OS: Linux
Toolchain Zephyr SDK 0.15.2
Zephyr 3.3.99

**Additional context**
Possible solutions include to initialize this variable in the case it is not used, as was the case in the past, or do the conditional checks associated with it on compile-time instead of runtime.

### PR DESCRIPTION
**Pull request by koffes**
current_dash may be used uninitialized

Fixes #57520

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Review by cfriedt — CHANGES_REQUESTED** [review]
@koffes - thanks for providing the fix. I'm actually unable to reproduce the issue using any of our existing testsuite configurations, but upon inspection, it's clear why this variable should be initialized.

If possible, could you or @Frodevan provide reproducibility details for the `tests/posix/getopt/` testsuite in #57520? E.g. was a static analysis tool used? Can the output of the static analyzer be added to the description?

**Review by cfriedt — CHANGES_REQUESTED** [review]
@koffes - one minor request just to ensure that we *do* have coverage for this in CI, and thank you for your patience.

Can you please add a second commit to this PR with the change below?
```diff
diff --git a/samples/bluetooth/iso_broadcast/sample.yaml b/samples/bluetooth/iso_broadcast/sample.yaml
index fdd0b34436..be80a0ad65 100644
--- a/samples/bluetooth/iso_broadcast/sample.yaml
+++ b/samples/bluetooth/iso_broadcast/sample.yaml
@@ -14,3 +14,12 @@ tests:
       - nrf52dk_nrf52832
     extra_args: OVERLAY_CONFIG=overlay-bt_ll_sw_split.conf
     tags: bluetooth
+  # to ensure that the issue reported in #57520 has coverage in CI
+  sample.bluetooth.iso_broadcast.getopt:
+    platform_allow: nrf5340_audio_dk_nrf5340_cpuapp
+    build_only: True
+    tags: bluetooth
+    extra_configs:
+      - CONFIG_COMPILER_WARNINGS_AS_ERRORS=y
+      - CONFIG_GETOPT=y
+      - CONFIG_GETOPT_LONG=y
```

cc @jakub-uC - would be great to identify why this was not caught by the `tests/posix/getopt` testsuite, since `CONFIG_COMPILER_WARNINGS_AS_ERRORS=y` is set by `twister`. Any ideas why that might be the case @nashif? It's a bit puzzling to me. Additional details [here](https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1536440041) and [here](https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1537519807)

File: lib/posix/getopt/getopt_long.c [hunk 1]
```diff
@@ -172,7 +172,7 @@ parse_long_options(struct getopt_state *state, char * const *nargv,
 {
 	char *current_argv, *has_equal;
 #ifdef GNU_COMPATIBLE
-	char *current_dash;
+	char *current_dash = "";
```
Comment (Thalley) [inline lib/posix/getopt/getopt_long.c:175]: Suggest to also remove 
```c
		default:
			current_dash = "";
break;
```
in the switch below, as that code is now redundant

Comment (jakub-uC) [reply lib/posix/getopt/getopt_long.c:175]: The root cause is explained in: https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1538130307

**Comment by jakub-uC** [comment]
The root cause is explained in: https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1538130307

File: lib/posix/getopt/getopt_long.c (hunk 1 above)
Comment (koffes) [reply lib/posix/getopt/getopt_long.c:175]: I have removed `current_dash = ""` but the default switch case is required by the code standard.

**Comment by cfriedt** [comment]
@koffes - again, sorry for the hoops, but it would be preferable if you could replace your second commit to `samples` with the change to `testcase.yaml` made in #56752.

With that, we have both the fix and the added test coverage in the same PR which is ideal for backports.

Thanks!

**Comment by koffes** [comment]
@cfriedt : I don't follow. In https://github.com/zephyrproject-rtos/zephyr/pull/56752 there is no mention of testcase.yaml.

**Comment by jakub-uC** [comment]
@koffes I think @cfriedt pointed out the wrong PR (the link is broken).

Please refer to this: #57652

**Comment by koffes** [comment]
There we go @jakub-uC and @cfriedt :)

**Review by cfriedt — APPROVED** [review]

**Review by jakub-uC — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**